### PR TITLE
dts: samples: rpc: convert to new devicetree macros

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Comment by galak** [comment]
Pulled commits out of #24728

File: samples/bluetooth/hci_rpmsg/src/main.c [hunk 1]
```diff
@@ -33,8 +33,13 @@
 LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 
 /* Configuration defines */
+#if !DT_HAS_CHOSEN(zephyr_ipc_shm)
+#error "Sample requires definition of shared memory for rpmsg"
```
Comment (joerchan) [inline samples/bluetooth/hci_rpmsg/src/main.c:37]: Maybe use `BUILD_ASSERT_MSG` instead?

File: samples/subsys/ipc/openamp_rsc_table/src/main_remote.c [hunk 2]
```diff
@@ -22,9 +22,14 @@ LOG_MODULE_REGISTER(openamp_rsc_table, LOG_LEVEL_DBG);
 #define RPMSG_CHAN_NAME	"rpmsg-client-sample"
 #define SHM_DEVICE_NAME	"shm"
 
-/* constant derivated from linker symbols */
-#define SHM_START_ADDR	DT_IPC_SHM_BASE_ADDRESS
-#define SHM_SIZE	(DT_IPC_SHM_SIZE * 1024)
+#if !DT_HAS_CHOSEN(zephyr_ipc_shm)
+#error "Sample requires definition of shared memory for rpmsg"
```
Comment (joerchan) [inline samples/subsys/ipc/openamp_rsc_table/src/main_remote.c:26]: Maybe use `BUILD_ASSERT_MSG` instead?

File: samples/bluetooth/hci_rpmsg/src/main.c (hunk 1 above)
Comment (ioannisg) [reply samples/bluetooth/hci_rpmsg/src/main.c:37]: From build-time perspective this gives the same result, IMHO.
Also, I would think a BUILD_ASSERT_MSG would be useful, if we wanted to check at build-time whether a value is not proper, but the check here is whether a node exists or not.

 @joerchan maybe I am missing sthg, so please, elaborate :)

File: samples/bluetooth/hci_rpmsg/src/main.c [hunk 3]
```diff
@@ -33,8 +33,13 @@
...
+#endif
 
-#define SHM_START_ADDR      (DT_IPC_SHM_BASE_ADDRESS + 0x400)
+#define SHM_NODE            DT_CHOSEN(zephyr_ipc_shm)
+#define SHM_BASE_ADDRESS    DT_REG_ADDR(SHM_NODE)
+#define SHM_START_ADDR      (SHM_BASE_ADDRESS + 0x400)
 #define SHM_SIZE            0x7c00
```
Comment (ioannisg) [inline samples/bluetooth/hci_rpmsg/src/main.c:43]: @galak since we are here let's fix this as well (it should be DT_REG_SIZE() - 0x400)

**Review by ioannisg — APPROVED** [review]

File: samples/bluetooth/hci_rpmsg/src/main.c (hunk 1 above)
Comment (joerchan) [reply samples/bluetooth/hci_rpmsg/src/main.c:37]: No strong reason, just thought BUILD_ASSERT_MSG would be more appropriate as a compile time check. But I see that #error is used in many places as well.

**Review by joerchan — APPROVED** [review]

File: samples/bluetooth/hci_rpmsg/src/main.c (hunk 3 above)
Comment (kapi-no) [reply samples/bluetooth/hci_rpmsg/src/main.c:43]: If so, please align it also here:
https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/bluetooth/hci/rpmsg_nrf53.c#L36

File: samples/bluetooth/hci_rpmsg/src/main.c (hunk 1 above)
Comment (ioannisg) [reply samples/bluetooth/hci_rpmsg/src/main.c:37]: Yes, that is acceptable, AFAIK

File: samples/bluetooth/hci_rpmsg/src/main.c (hunk 3 above)
Comment (galak) [reply samples/bluetooth/hci_rpmsg/src/main.c:43]: Will let that be a followup PR.

**Pull request merged**